int32 activeNode
float64 target1_x
float64 target1_y
float64 target2_x
float64 target2_y
float64 target3_x
float64 target3_y